master/internal/experiment.go

```diff
@@ -699,73 +621,57 @@ func (e *internalExperiment) processOperations(
 	defer e.snapshotAndSave()
 
 	updatedTrials := make(map[model.RequestID]bool)
-	for _, operation := range ops {
-		e.syslog.Debugf("handling searcher op: %v", operation)
-		switch op := operation.(type) {
+	for _, action := range actions {
+		e.syslog.Debugf("handling searcher action: %v", action)
+		switch action := action.(type) {
 		case searcher.Create:
-			_, ok := e.trials[op.RequestID]
+			_, ok := e.trials[action.RequestID]
 			if ok {
-				e.syslog.Errorf("trial %s already exists", op.RequestID)
+				e.syslog.Errorf("trial %s already exists", action.RequestID)
 				continue
 			}
 
-			continueFromTrialID, closed := e.handleContinueExperiment(op.RequestID)
+			continueFromTrialID, closed := e.handleContinueExperiment(action.RequestID)
 			if closed {
 				continue
 			}
+			state := experiment.TrialSearcherState{Create: action}
+			e.TrialSearcherState[action.RequestID] = state
 
-			checkpoint, err := e.checkpointForCreate(op)
-			if err != nil {
-				e.updateState(model.StateWithReason{
-					State: model.StoppingErrorState,
-					InformationalReason: fmt.Sprintf(
-						"hp search unable to get checkpoint for new trial with error %v", err),
-				})
-				e.syslog.Error(err)
-				continue
-			}
 			config := schemas.Copy(e.activeConfig)
-			state := experiment.TrialSearcherState{Create: op, Complete: true}
-			e.TrialSearcherState[op.RequestID] = state
 
 			clonedSpec, err := e.taskSpec.Clone()
 			if err != nil {
 				e.syslog.WithError(err).Error("failed to create trial")
-				e.trialClosed(op.RequestID, ptrs.Ptr(model.Errored))
+				e.trialExited(action.RequestID, ptrs.Ptr(model.Errored))
 				continue
 			}
 
 			t, err := newTrial(
-				e.logCtx, trialTaskID(e.ID, op.RequestID), e.JobID, e.StartTime, e.ID, e.State,
-				state, e.rm, e.db, config, checkpoint, clonedSpec, e.generatedKeys, false,
-				nil, continueFromTrialID, e.TrialClosed,
+				e.logCtx, trialTaskID(e.ID, action.RequestID), e.JobID, e.StartTime, e.ID, e.State,
+				state, e.rm, e.db, config, e.warmStartCheckpoint, clonedSpec, e.generatedKeys, false,
```

Comment:
i don't quite follow why we removed `checkpointForCreate` and the work it did. i tried to hold opinions until i read `trial.go` but i dont see any logic moved there.

does python code do it now or something?

Reply:
it wasn't being used AFAICT, pretty sure we needed it at some point to support PBT and custom searcher, but both of those are dead now.